### DISCUSSION TITLE
WebSocket Support

### DISCUSSION THREAD

**Discussion by amz202** [br-g/openf1#39] [Q&A] [answered]
Hello, I am a newbie and want to make an android app using this api. I haven't worked with fetching live data with real time updates before so don't mind this basic question

My question is, does this API support the WebSockets? I plan to use that for real time data updates in the app. Thanks.

**Comment by br-g** [comment, marked as answer]
Hello, 
The API does not support WebSockets. Sorry!
However, it is possible with the current system to fetch real-time data at regular intervals

**Reply by amz202** [reply]
Oh alright! Thanks for responding.

**Reply by thefeli73** [reply]
@br-g Hi, is the best option for real time data to just call the api every 500ms basically? I feel like a websocket stream with new data as it comes in would be cheaper on your end and lead to a better developer experience?

**Reply by br-g** [reply]
Eventually, we may have a web socket API, but it is not the current priority

**Reply by rtoscani** [reply]
I was looking for this also!

Maybe we can expose a SSE endpoint on the API and communicate the ingestion process with the API through a Redis queue. Should be simple and cheap to do so.

**Reply by bregnvig** [reply]
@br-g I'm considering doing a firebase project, with firestore for as realtime database for OpenF1 data. So it would be OK to call every 500ms to fetch latest data? (During sessions of course)
